internal/addon/manifests/charts/mcoa/charts/metrics/templates/clusterrolebinding.yaml

```
@@ -0,0 +1,28 @@
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: open-cluster-management:metrics-addon:agent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: open-cluster-management:metrics-addon:agent
+  rules:
+    - apiGroups: [""]
+      resources:
+        - nodes
+        - nodes/proxy
+        - services
+        - endpoints
+        - pods
+      verbs: ["get", "list", "watch"]
+    - apiGroups:
+        - extensions
+      resources:
+        - ingresses
+      verbs: ["get", "list", "watch"]
+    - nonResourceURLs: ["/federate"]
+      verbs: ["get"]
+subjects:
+  - kind: ServiceAccount
+    name: metrics-addon-sa
+    namespace: { { .AddonInstallNamespace } }
```

Forgot to mention in my morning review but to allow for the metrics chart to be disabled you will have to each resource the following `{{- if .Values.enabled }} ... {{- end }}`. We need to do this way because the addon is not able to recognize the dependencies in the `Chart.yaml` and enable/disable things accordingly

```suggestion
{- if .Values.enabled }}
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: open-cluster-management:metrics-addon:agent
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: open-cluster-management:metrics-addon:agent
  rules:
    - apiGroups: [""]
      resources:
        - nodes
        - nodes/proxy
        - services
        - endpoints
        - pods
      verbs: ["get", "list", "watch"]
    - apiGroups:
        - extensions
      resources:
        - ingresses
      verbs: ["get", "list", "watch"]
    - nonResourceURLs: ["/federate"]
      verbs: ["get"]
subjects:
  - kind: ServiceAccount
    name: metrics-addon-sa
    namespace: { { .AddonInstallNamespace } }
{{- end }} 
```